front/app/containers/Admin/projectFolders/containers/settings/ProjectFolderForm/index.tsx

```diff
@@ -335,6 +335,7 @@ const ProjectFolderForm = ({ mode, projectFolderId }: Props) => {
                 },
               }
             );
+            setSubmitState('success');
```

Comment:
So upload failed because the app "thought" it didn't succeed?

I think it would be nice (and more robust) if we would only have to write setSubmitState('success') once, for example at the end of the  `saveForm` method. But looks complicated, this is such a long method 😮

Do you think this could be realized by adding `return` under each `setSubmitState` for errors?

Reply:
>So upload failed because the app "thought" it didn't succeed?

Sort of - we just weren't properly updated the submitState after everything succeeded (we left it on the "Loading" state when we needed to update it to "Success").

>But looks complicated, this is such a long method 
Indeed - not nice code in this file - it's a file which is quite old and messy :sweat_smile: